File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        for(int i=0;i<intervals.size();i++)
        {
            reverse(intervals[i].begin(),intervals[i].end());
        }

        sort(intervals.begin(),intervals.end());
        int time=INT_MIN;

        int n=intervals.size();
        int meetings=0;

        

        for(int i=0;i<n;i++)
        {
            if(time<=intervals[i][1])
            {
                meetings++;
                time=intervals[i][0];
            }
        }
    //    cout<<meetings;

        return n-meetings;
    }
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution(object):
    def eraseOverlapIntervals(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: int
        """

        intervals.sort()
        #print(intervals)

        start=intervals[0][0]
        end=intervals[0][1]
        j = 0
        
        overlappingInter = []
        ans=0

        n=len(intervals)
        for i in range(1, n):
            if(intervals[i][0]<end):
                
                if(intervals[i][1]<end):
                    #delete previous elemnt
                    overlappingInter.append(intervals[j])
                    j=i
                    start=intervals[j][0]
                    end=intervals[j][1]
                    #inc the count
                else:
                    # delete current element
                    overlappingInter.append(intervals[i])
            else:
                j=i
                start=intervals[j][0]
                end=intervals[j][1]

        print(overlappingInter)
        return len(overlappingInter)
        
                    

        